src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2768,7 +2745,7 @@ private bool ShouldInferDeclarationExpressionValEscape(BoundExpression argument,
                 _ => null
             };
             if (symbol is SourceLocalSymbol local &&
-                GetLocalScopes(local).ValEscapeScope == CallingMethodScope)
+                GetLocalScopes(local).ValEscapeScope.IsCallingMethod)
```

Comment:
At some places we use `scope.IsXYZ` and in others we use `scope.Equals(Lifetime.XYZ)`. Do we even need the `IsXYZ` properties; wouldn't it be better to use `.Equals` everywhere? (When I see the property I have to wonder if it returns `true` for multiple possible scopes like `IsReturnable` whereas `.Equals` is clear that we are checking for exactly one scope.)

Reply:
The main challenge without `bool IsReturnOnly` prop, for example, is: how to check in a pattern, if a lifetime equals some particular well-known lifetime. These could be moved to 'when' clauses or some such, though.

```cs
Lifetime li = ...;
li switch { { IsReturnOnly: true } => ... };
```